### DISCUSSION TITLE
Docker install failed

### DISCUSSION THREAD

**Discussion by ankyfive** [Budibase/budibase#3241] [Help] [answered]
Hello,
I try to install budibase with docker.

[root@abcd:/var/lib/docker/budibase]# ll
-rw-r--r--  1 root root 3903 Nov  3 15:29 docker-compose.yaml
-rw-r--r--  1 root root 4273 Nov  3 15:29 envoy.yaml
-rw-r--r--  1 root root  589 Nov  3 15:29 hosting.properties

[root@abcd:/var/lib/docker/budibase]# docker-compose --file hosting.properties up
ERROR: yaml.parser.ParserError: expected '<document start>', but found '<scalar>'
  in "./hosting.properties", line 6, column 1

Don't understand why it does not work.
Any help would by appreciated.

Thanks

**Comment by shogunpurple** [comment, marked as answer]
Hi @ankyfive the flag is `--env-file` rather than `--file`.  

`--file` is when you want to point to an alternate `docker-compose.yaml` configuration.

Thanks!

**Reply by ankyfive** [reply]
Hi shogunpurple
Thanks for your reply.
I tried --file cause --env-file is not recognized as a docker-copose option
Sorry I am new to docker...

**Reply by shogunpurple** [reply]
`env-file` is only available in later versions of `docker-compose`. You have 2 options:

## Upgrade docker-compose
`pip install docker-compose`
`docker-compose --env-file hosting.properties up -d`

## Move hosting.properties to .env file and run again
This will allow you to use the `.env` file without specifying `--env-file`
`mv hosting.properties .env`
`docker-compose up -d`

**Reply by ankyfive** [reply]
Thanks for the info, but when I run the command I get:

[root@abcd:/var/lib/docker/budibase]# docker-compose --env-file hosting.properties up -d
Define and run multi-container applications with Docker.

Usage:
  docker-compose [-f <arg>...] [options] [COMMAND] [ARGS...]
  docker-compose -h|--help

Options:
  -f, --file FILE             Specify an alternate compose file
                              (default: docker-compose.yml)
  -p, --project-name NAME     Specify an alternate project name
                              (default: directory name)
  --verbose                   Show more output
  --log-level LEVEL           Set log level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
  --no-ansi                   Do not print ANSI control characters
  -v, --version               Print version and exit
  -H, --host HOST             Daemon socket to connect to

  --tls                       Use TLS; implied by --tlsverify
  --tlscacert CA_PATH         Trust certs signed only by this CA
  --tlscert CLIENT_CERT_PATH  Path to TLS certificate file
  --tlskey TLS_KEY_PATH       Path to TLS key file
  --tlsverify                 Use TLS and verify the remote
  --skip-hostname-check       Don't check the daemon's hostname against the
                              name specified in the client certificate
  --project-directory PATH    Specify an alternate working directory
                              (default: the path of the Compose file)
  --compatibility             If set, Compose will attempt to convert deploy
                              keys in v3 files to their non-Swarm equivalent

Commands:
  build              Build or rebuild services
  bundle             Generate a Docker bundle from the Compose file
  config             Validate and view the Compose file
  create             Create services
  down               Stop and remove containers, networks, images, and volumes
  events             Receive real time events from containers
  exec               Execute a command in a running container
  help               Get help on a command
  images             List images
  kill               Kill containers
  logs               View output from containers
  pause              Pause services
  port               Print the public port for a port binding
  ps                 List containers
  pull               Pull service images
  push               Push service images
  restart            Restart services
  rm                 Remove stopped containers
  run                Run a one-off command
  scale              Set number of containers for a service
  start              Start services
  stop               Stop services
  top                Display the running processes
  unpause            Unpause services
  up                 Create and start containers
  version            Show the Docker-Compose version information

**Reply by shogunpurple** [reply]
still seems like your `docker-compose` is out of date. Could you paste the output of:

`docker-compose --version`

**Reply by ankyfive** [reply]
This is the version :
docker-compose version 1.21.0, build unknown

I tried:
mv hosting.properties .env
docker-compose up -d

It downloaded a lot of things but after a while :

ERROR: Get "https://budibase.docker.scarf.sh/v2/": Forbidden

**Reply by shogunpurple** [reply]
hm, seems like scarf is maybe getting blocked by your company firewall.

Please try the suggestion in this comment https://github.com/Budibase/budibase/issues/2759#issuecomment-938043128

and run again.

**Reply by ankyfive** [reply]
Yes you are right we allowed the URL in the firewall and it work !
Thanks for your help :-)
